### PR TITLE
Updated bower Dependencies

### PR DESCRIPTION
**Pull request by muskankhedia**
## Description
All the compatible libraries in the bower.json have been updated. The list of the updated libraries with their versions are as follows:

| Package Name        | Previous Version           | Updated Version  |
| ------------- |:-------------:| -----:|
| angular     | 1.5.0 | 1.7.9 |
| angular-resource  | 1.5.0 | 1.7.9 |
| angular-route     | 1.5.0 | 1.7.9 |
| angular-mocks     | 1.5.0 | 1.7.9 |
| angular-animate     | 1.5.0 | 1.7.9 |
| angular-touch     | 1.5.0 | 1.7.9 |
| requirejs     | 2.2.0 | ^2.3.6 |
| require-css      | 0.1.8      |   ^0.1.10 |
| angular-translate | 2.7.2  |  2.18.2   |
| angular-translate-loader-static-files | 2.7.2      |   2.18.2 |
| angular-sanitize     | 1.5.0 | 1.7.9 |
| ckeditor | 4.5.7   |   ^4.14.0 |
| angular-dynamic-locale |    | ^0.1.37 |
| angular-chosen-localytics |  ^1.4.0     | ^1.9.3    |
| almond | ^0.3.2    |  ^0.3.3  |

## Related issues and discussion
#3209 

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Validate the JS and HTML files with `grunt validate` to detect errors and potential problems in JavaScript code.

- [x] Run the tests by opening `test/SpecRunner.html` in the browser to make sure you didn't break anything.

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `community-app/Contributing.md`.

### CODE REVIEW

**Comment by abhaychawla** [comment]
@muskankhedia Are we able to update angular version to 1.7 as well?

**Comment by muskankhedia** [comment]
@abhaychawla One of the packages `angular-sanitize` is broken in the future release and there is an error as well `angular.lowercase is not a function` when we update it to `angular 1.7.9`. The description of the error has been explained in the [link](https://github.com/angular/angular.js/issues/16563). Also, all the angular packages should be in the same version for the best performance and utilisation. So I didn't upgrade the modules that would anyway break the application and also didn't update any of the angular packages.

**Comment by abhaychawla** [comment]
@muskankhedia Are we able to use any of the workarounds mentioned at https://github.com/textAngular/textAngular/issues/1608 to tackle this?

**Comment by muskankhedia** [comment]
@abhaychawla Thanks for the link, I found a solution to tackle the same. Now, it's working fine after upgrading the angular modules as well but needs changes in some files based on upgraded functionality. Will make the changes and update the PR soon.

**Comment by abhaychawla** [comment]
@muskankhedia They are minor changes?

**Comment by muskankhedia** [comment]
@abhaychawla Yes somehow, needs changes in some files based on the updated modules.

**Comment by abhaychawla** [comment]
@muskankhedia How soon can we make those changes? Can we target it by tomorrow?

**Comment by muskankhedia** [comment]
Yaa sure. I will try to complete it by tomorrow.

**Comment by muskankhedia** [comment]
@abhaychawla I tried fixing up the changes as mentioned in the [docs](https://code.angularjs.org/1.7.0-rc.0/docs/guide/migration#migrating-from-1-6-to-1-7) but still facing some errors with the Authentication and the Notification modules. Can you please have a look at the same?

**Comment by abhaychawla** [comment]
@muskankhedia Please share the screenshot of the error here once.

**Comment by muskankhedia** [comment]
@abhaychawla This is the screenshot of the homepage with the console after making the changes.

Screenshot from 2020-05-26 20-21-15

**Comment by edcable** [comment]
@abhaychawla and @muskankhedia have we made any progress on the errors that you were looking into?

**Comment by abhaychawla** [comment]
@edcable Still need to look into it, I have been a bit preoccupied with office work. I will try to look at by next week.

File: app/scripts/controllers/main/MainController.js [hunk 1]
```diff
@@ -2,7 +2,8 @@
     mifosX.controllers = _.extend(module, {
         MainController: function (scope, location, sessionManager, translate, $rootScope, localStorageService, keyboardManager, $idle, tmhDynamicLocale,
                   uiConfigService, $http) {
-            $http.get('release.json').success(function(data) {
+            $http.get('release.json').then(function(data) {
```
Comment (Jov03) [inline app/scripts/controllers/main/MainController.js]:
```suggestion
            $http.get('release.json').then(function(response) {
            var data = response.data;
```

Refer : https://docs.angularjs.org/guide/migration#-http-

Note: The method signatures and return values are different. In Angular v1.5 success method used to return the following data:
`$http(...).
success(function onSuccess(data, status, headers, config) {
  // Handle success
  ...
}).
error(function onError(data, status, headers, config) {
  // Handle error
  ...
});`

but now it returns a single object containing the above mentioned property :

```
$http(...).
  then(function onSuccess(response) {
    // Handle success
    var data = response.data;
    var status = response.status;
    var statusText = response.statusText;
    var headers = response.headers;
    var config = response.config;
    ...
  }).
  catch(function onError(response) {
    // Handle error
    var data = response.data;
    var status = response.status;
    var statusText = response.statusText;
    var headers = response.headers;
    var config = response.config;
    ...
  });
```

File: app/scripts/controllers/main/MainController.js [hunk 2]
```diff
@@ -16,7 +17,8 @@
 
             if(!scope.islogofoldernamefetched && $rootScope.tenantIdentifier && $rootScope.tenantIdentifier != "default"){
                 scope.islogofoldernamefetched = true;
-                $http.get('scripts/config/LogoConfig.json').success(function(datas) {
+                $http.get('scripts/config/LogoConfig.json').then(function onSuccess(datas) {
```
Comment (Jov03) [inline app/scripts/controllers/main/MainController.js]: Refer : https://docs.angularjs.org/guide/migration#-http-

File: app/scripts/controllers/reports/XBRLController.js [hunk 3]
```diff
@@ -12,7 +12,7 @@
                     method: 'GET',
                     url: $rootScope.hostUrl + API_VERSION + '/mixmapping'
                 })
-                    .success(function (data) {
+                    .then(function (data) {
```
Comment (Jov03) [inline app/scripts/controllers/reports/XBRLController.js]: Refer : https://docs.angularjs.org/guide/migration#-http-

File: app/scripts/controllers/reports/XBRLController.js [hunk 4]
```diff
@@ -95,11 +97,13 @@
                 http({
                     method: 'GET',
                     url: $rootScope.hostUrl + API_VERSION + '/mixreport?startDate=' + scope.startDate + '&endDate=' + scope.endDate
-                }).success(function (data) {
+                }).then(function (data) {
```
Comment (Jov03) [inline app/scripts/controllers/reports/XBRLController.js]: Refer : https://docs.angularjs.org/guide/migration#-http-

File: app/scripts/models/LoggedInUser.js [hunk 5]
```diff
@@ -7,10 +7,12 @@
 
             this.getHomePageIdentifier = function () {
                 var role = _.first(data.selectedRoles || data.roles);
-                if (role.id in mifosX.models.roleMap) {
-                    return mifosX.models.roleMap[role.id];
-                } else {
-                    return 'default';
+                if (role) {
```
Comment (Jov03) [inline app/scripts/models/LoggedInUser.js]: Revert this if block. `data.roles` and 'data.selectedRoles' were undefined due to mistakes in `$http.then` method.

File: app/scripts/modules/notificationWidget.js [hunk 6]
```diff
@@ -162,7 +162,11 @@ angular.module('notificationWidget', [])
                         for(var i = 0; i < data.length; i++) {
                             //console.log(data[i]);
                             var jsonErrors = JSON.parse(data[i].body);
-                            var valErrors = jsonErrors.errors;
+                            if (jsonErrors) {
```
Comment (Jov03) [inline app/scripts/modules/notificationWidget.js]: revert this if block and check the handling in `$http.then` method

File: app/scripts/modules/notificationWidget.js [hunk 7]
```diff
@@ -192,11 +196,13 @@ angular.module('notificationWidget', [])
                                  * Update user password api call won't return errors array,
                                  * if user enters a password which is used previously
                                  */
-                                if (jsonErrors.userMessageGlobalisationCode) {
-                                    var errorObj = new Object();
-                                    errorObj.code = jsonErrors.userMessageGlobalisationCode;
-                                    errorArray[arrayIndex] = errorObj;
-                                    arrayIndex++;
+                                if(jsonErrors) {
```
Comment (Jov03) [inline app/scripts/modules/notificationWidget.js]: revert this if block and check the handling in $http.then method

File: app/scripts/services/UIConfigService.js [hunk 8]
```diff
@@ -15,18 +15,30 @@
...
+            //     }).catch(function(e){
+            //         console.log("Configuration file not found");
+            //     });
+
+            // };
             this.init  = function(scope) {
                 var deferred = $q.defer();
-                $http.get('scripts/config/UIconfig.json').success(function(data) {
+                $http.get('scripts/config/UIconfig.json')
+                .then(function(data) {
```
Comment (Jov03) [inline app/scripts/services/UIConfigService.js]: Refer : https://docs.angularjs.org/guide/migration#-http-

**Review by Jov03 — CHANGES_REQUESTED** [review]
@muskankhedia you need to refer https://docs.angularjs.org/guide/migration . To make sure you have made the changes required to go from angular v1.5 to angular v1.6 then angular v1.6 to angular v1.7 

Right now the entire community app is not working in your PR due to 2 reasons:

1. incorrect handling of `$http.then`  method 
2. In angular v1.6  the default hash-prefix in url has been changed to (!)  so you need to add 
`$locationProvider.hashPrefix('');` in routes.js to set it back to empty string.
Refer : https://docs.angularjs.org/guide/migration#commit-aa077e8

@edcable is it necessary to update other dependencies too besides angular? If it is @muskankhedia you would need  to refer the changelog of each and every dependency version you are upgrading and make the necessary changes in code where that dependecy is used.

**Comment by edcable** [comment]
@Jov03 for security reasons, we were trying to ensure that the dependencies were updates. @spider0061 are you able to assist at all here? Is this at all related to #3247

**Comment by muskankhedia** [comment]
@edcable I am working to make the updates as reviewed by @Jov03 . I was working on some web-app issues and gsoc final documentation so couldn't get time much time for this PR. I will try to update it in a day or two.

**Comment by spider0061** [comment]
> @Jov03 for security reasons, we were trying to ensure that the dependencies were updates. @spider0061 are you able to assist at all here? Is this at all related to #3247

The issue with #3247 is  because of phantomjs. I am looking into it.

**Review by Jov03 — APPROVED** [review]
Thanks @muskankhedia !! This PR looks good.

**Pull request merged**